http-webhook.md

```diff
@@ -0,0 +1,339 @@
+
+The `WebHook-Request-Origin` header requests permission to send notifications
+from this sender, and contains a DNS expression that identifies the sending
+system, for example "eventemitter.example.com". The value is meant to summarily
+identify all sender instances that act on the behalf of a certain system, and
+not an individual host.
+
+After the handshake and when permission is granted, the sender MUST use the
+`Origin` request header for each delivery request, with the value matching that
+of this header.
+
+For example:
+
+``` text
+WebHook-Request-Origin: eventemitter.example.com
+```
+
+#### 4.1.1.2. WebHook-Request-Rate
+
+The `WebHook-Request-Rate` header MAY be included in the request and asks for
+permission to send notifications from this sender at the specified rate. The
+value is the string representation of a positive integer number greater than
+zero and expresses the request rate in "requests per minute".
+
+For example, the following header asks for permission to send 120 requests per
+minute:
+
+``` text
+WebHook-Request-Rate: 120
+```
+
+### 4.1.2 Validation response
+
+If and only if the delivery target does allow delivery of the events, it MUST
+reply to the request by including the `WebHook-Allowed-Origin` and
+`WebHook-Allowed-Rate` headers.
+
+If it does not allow delivery of the events or does not expect delivery of
+events and nevertheless handles the HTTP OPTIONS method, the existing response
+ought not to be interpreted as consent, and therefore the handshake cannot
+rely on status codes.
+
+The response SHOULD include the [Allow][Allow] header indicating the [POST][POST]
+method being permitted. Other methods MAY be permitted on the resource, but
+their function is outside the scope of this specification.
+
+#### 4.1.1.2. WebHook-Allowed-Origin
+
+The `WebHook-Allowed-Origin` header MUST be returned when the delivery target
+agrees to notification delivery by the origin service. Its value MUST either be
+the origin name supplied in the `WebHook-Request-Origin` header, or a singular
+asterisk character ('*'), indicating that the delivery target supports
+notifications from all origins.
+
+``` text
+WebHook-Allowed-Origin: eventemitter.example.com
+```
+
+or
+
+``` text
+WebHook-Request-Origin: *
+```
+
+#### 4.1.1.2. WebHook-Allowed-Rate
+
+The `WebHook-Allowed-Rate` header MUST be returned if the request contained
+teh `WebHook-Request-Rate`, otherwise it SHOULD be returned.
+
+The header grants permission to send notifications at the specified rate. The
+value is either an asterisk character or the string representation of a
+positive integer number greater than zero. The asterisk indicates that there is
+no rate limitation. An integer number expresses the permitted request rate in
+"requests per minute". For request rates exceeding the granted notification
+rate, the sender ought to expect request throttling. Throttling is indicated
+by requests being rejected using HTTP status code [429][429].
```

Comment:
HTTP 429 Too Many Requests

Reply:
I don't really see what is wrong here, it's referring to the spec here so looks ok?

Reply:
Done (adding the reason phrase for clarity)